test/modules/EdgeHubRestartTester/src/DirectMethodEdgeHubConnectorTest.cs

```diff
@@ -25,10 +25,12 @@ class DirectMethodEdgeHubConnectorTest : IEdgeHubConnectorTest
 
         public DirectMethodEdgeHubConnectorTest(
             Guid batchId,
-            ILogger logger)
+            ILogger logger,
+            ModuleClient dmModuleClient = null)
```

Comment:
why dmModuleClient allows null and optional?  module client should be created in main program and clean in there as well.  Connector test do not need to implement IDisposable.